scalyr_agent/monitor_utils/k8s.py

```diff
@@ -1061,18 +1091,23 @@ def update_cache( self, run_state ):
 
             try:
                 current_time = time.time()
-                if local_state.batch_pod_updates:
-                    self._pods.update( local_state.k8s, local_state.node_filter, 'Pod' )
+                if not self._state.cache_config.use_controlled_warmer:
```

Comment:
IIUC all of the following logic should be avoided when using warmer. 
Instead, the entire pod cache should be invalidated.